Java/core/src/main/java/com/amazon/randomcutforest/tree/CompactRandomCutTreeDouble.java

```diff
@@ -55,21 +56,23 @@
     public static final short NULL = -1;
 
     private final Random random;
-    private final int maxSize;
+    protected final short maxSize;
     protected final NodeStore internalNodes;
     protected final LeafStore leafNodes;
     protected final IPointStore<double[]> pointStore;
     protected short rootIndex;
+    private final CompactNodeViewDouble usher;
 
     public CompactRandomCutTreeDouble(int maxSize, long seed, IPointStore<double[]> pointStore) {
         checkArgument(maxSize > 0, "maxSize must be greater than 0");
         checkNotNull(pointStore, "pointStore must not be null");
-        this.maxSize = maxSize;
+        this.maxSize = (short) maxSize;
```

Comment:
Can we make the constructor argument a short so we don't have to cast it?

Reply:
fixed